libappstream-glib/as-store.c

```diff
@@ -3157,6 +3221,70 @@ as_store_load (AsStore *store,
 	return TRUE;
 }
 
+static void
+store_load_thread (GTask *task,
+		   gpointer source_object,
+		   gpointer task_data,
+		   GCancellable *cancellable)
+{
+	AsStore *store = source_object;
```

Comment:
Do we need a `AS_STORE()` here?

Reply:
Good question. (void*) pointers don't require a cast and this function is only called internally, so after adding g_return(_val)_if_fail to the public methods, we should be safe ... unless the checks are disabled via G_DISABLE_CHECKS.

Do you want me to add the cast for that case?

Reply:
Sure, I added it now.